src/Form/WebdamUpload.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\media_webdam\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\media_webdam\WebdamInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Implements form to save media files and upload them to Webdam.
+ *
+ * @package Drupal\media_webdam\Form
+ */
+class WebdamUpload extends FormBase {
+
+  /**
+   * Drupal\media_webdam\WebdamInterface definition.
+   *
+   * @var \Drupal\media_webdam\WebdamInterface
+   */
+  protected $webdam;
+
+  /**
+   * The storage handler class for files.
+   *
+   * @var \Drupal\file\FileStorage
+   */
+  protected $fileStorage;
+
+  /**
+   * The EntityManagerInterface.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * WebdamUpload constructor.
+   *
+   * @param \Drupal\media_webdam\WebdamInterface $webdam
+   *   The Webdam elements.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityManager
+   *   The EntityManagerInterface.
+   */
+  public function __construct(WebdamInterface $webdam, EntityTypeManagerInterface $entityManager) {
+    $this->webdam = $webdam;
+    $this->entityManager = $entityManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('media_webdam.webdam'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'webdam_upload';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['upload_media'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Upload media'),
+    ];
+    $form['upload_media']['managed_file'] = [
+      '#type' => 'managed_file',
+      '#title' => $this->t('Media Upload'),
+      '#description' => $this->t('Select a file to Upload. Max upload size: 1MB'),
+      '#upload_location' => 'temporary://',
+      '#upload_validators' => [
+        'file_validate_extensions' => ['gif png jpg jpeg mp3 mp4 mkv'],
+      ],
+      '#multiple' => FALSE,
+      '#required' => TRUE,
+    ];
+    $form['upload_media']['webdam_folder'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Webdam folder'),
+      '#description' => $this->t('Please select a Webdam folder to store your file'),
+      '#options' => $this->webdam->getFlattenedFolderList(),
```

Comment:
Can you add a @TODO here to change this from the flattened folder list to something a little nicer?

Reply:
Sure. Do you mean the method name, right?

Reply:
Yeah. We don't have an alternative right now, but when we do, we should use it.